modules/database/database.bal

```diff
@@ -588,22 +603,29 @@ public isolated transactional function getDataTypesSummary(int experimentId) ret
     return dataSummary;
 }
 
-public isolated transactional function getDataList(int experimentId, boolean all = true, int 'limit = 100, int offset = 0) returns ExperimentDataFull[]|error {
-    var baseQuery = `SELECT dataId, experimentId, name, version, location, type, contentType 
+public isolated transactional function getDataList(int experimentId, string? search, boolean all = true, int 'limit = 100, int offset = 0, int? 'ascending = 1) returns ExperimentDataFull[]|error {
+    log:printDebug(">>>>>" + 'ascending.toString());
+    sql:ParameterizedQuery baseQuery = `SELECT dataId, experimentId, name, version, location, type, contentType 
                      FROM ExperimentData WHERE experimentId=${experimentId} `;
-    var baseQuerySuffix = `ORDER BY name ASC, version DESC 
-                           LIMIT ${'limit} OFFSET ${offset};`;
-
-    stream<ExperimentDataFull, sql:Error?> experimentData;
-    if all {
-        experimentData = experimentDB->query(check new ConcatQuery(baseQuery, baseQuerySuffix));
-    } else {
-        var extraFilter = `AND version=(SELECT MAX(t2.version)
+    if search != () && search != "" {
+        string searchString = "%" + search + "%";
+        baseQuery = sql:queryConcat(baseQuery, `AND ((name LIKE ${searchString}) OR (type LIKE ${searchString}) OR (contentType LIKE ${searchString})) `);
+    }
+    if !all {
+        baseQuery = sql:queryConcat(baseQuery, `AND version=(SELECT MAX(t2.version)
                                 FROM ExperimentData AS t2 
-                                WHERE ExperimentData.name=t2.name AND t2.experimentId=${experimentId}) `;
-        experimentData = experimentDB->query(check new ConcatQuery(baseQuery, extraFilter, baseQuerySuffix));
+                                WHERE ExperimentData.name=t2.name AND t2.experimentId=${experimentId}) `);
+    }
+    if ('ascending != () && 'ascending == 1) || 'ascending == () {
```

Comment:
These checks should happen in the API resource function in main.bal. Here all values should already be normalized and vetted.